src/test/groovy/sdkman/stubs/WebServiceStub.groovy

```diff
@@ -30,10 +33,17 @@ class WebServiceStub {
 	}
 
 	static primeDownloadFor(String host, String candidate, String version, String platform) {
+		primeDownloadFor(host, candidate, version, platform, [:])
+	}
+	
+	static primeDownloadFor(String host, String candidate, String version, String platform, Map<String, String> map) {
 		def binary = (candidate == "java") ? "jdk-${version}-${platform}.tar.gz" : "${candidate}-${version}.zip"
+		map["Location"] = "${host}/${binary}"
```

Comment:
The mutation of this headers map here feels a bit wrong. Is it possible to do this at the time of map creation in the step def explicitly when it is created?

Reply:
I refactored it a bit